application-task-ui/src/main/resources/TaskManager/EditorContentUpdater.xml

```diff
@@ -0,0 +1,543 @@
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'diff-dom'], function ($, diffDOM) {
+  let htmlDecode = function(content) {
+    let doc = new DOMParser().parseFromString(content, "text/html");
+    return doc.documentElement.textContent;
+  };
+  let handleCKEWysiwyg = function (content, editor) {
+    let currentHTMLContent = CKEDITOR.plugins.xwikiSource.getFullData(editor);
+    let convertURL = XWiki.currentDocument.getURL('get', $.param({
+      sheet: 'CKEditor.HTMLConverter',
+      outputSyntax: 'plain',
+      language: $('html').attr('lang') || '',
+      formToken: document.documentElement.dataset.xwikiFormToken || ''
+    }));
+    $.ajax({
+      url: convertURL,
+      method: 'POST',
+      data: {
+        sourceSyntax: 'xwiki/2.1',
+        // might be true if editing inline
+        stripHTMLEnvelope: editor.elementMode === CKEDITOR.ELEMENT_MODE_INLINE,
+        fromHTML: false,
+        toHTML: true,
+        text: content
+      }
+    })
+      .done(function (content) {
+        let dd = new diffDOM.DiffDOM();
+        let newContent;
+        if (editor.elementMode === CKEDITOR.ELEMENT_MODE_INLINE) {
+          let tmpServerBody = $('&lt;body&gt;&lt;/body&gt;').append(content);
+          let tmpCurrentBody = $('&lt;body&gt;&lt;/body&gt;').append(currentHTMLContent);
+          let diff = dd.diff(diffDOM.nodeToObj(tmpCurrentBody[0]), diffDOM.nodeToObj(tmpServerBody[0]))
+          dd.apply(tmpCurrentBody[0], diff);
+          newContent = tmpCurrentBody[0].innerHTML;
+        } else {
+          let currentDOM = $(new DOMParser().parseFromString(currentHTMLContent, 'text/html'));
+          let serverDOM = $(new DOMParser().parseFromString(content, 'text/html'));
+          // Compare the content of the body and apply the diff. We want to preserve the head.
+          let diffAgainst = serverDOM.find('body')[0];
+          let diffFrom = currentDOM.find('body')[0];
+          let diff = dd.diff(diffDOM.nodeToObj(diffFrom), diffDOM.nodeToObj(diffAgainst))
+          dd.apply(diffFrom, diff);
+          newContent = new XMLSerializer().serializeToString(currentDOM[0]);
+        }
+        let cursorRanges = editor.getSelection().getRanges();
+        editor.setData(newContent);
```

Comment:
Have you tried to apply the patch directly on the edited content, instead of setting the editor data as string? See what the real-time editor does https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-wysiwyg/xwiki-platform-realtime-wysiwyg-webjar/src/main/webjar/wysiwygEditor.js#L429 .